src/views/CharacterBuild.js

```diff
@@ -11,86 +11,78 @@ const useStyles = makeStyles({
-    crit_damage: 194.6,
-  };
-
-  //cleans up inputted build (above)
-  const charBuild = new CharacterClass(build);
-  const baseDamage = CharacterBaseDamage(charBuild);
 
   return (
-    <>
-      <Box style={{ display: "flex", alignItems: "center" }}>
-        <Box
-          style={{
-            justifyContent: "flex-start",
-            minWidth: "250px",
-            textAlign: "right"
-          }}
-          marginRight="20px"
-        >
-          character_name
-        </Box>
-        <Box style={{ justifyContent: "flex-end", minWidth: "100px" }}>
-          {get(charBuild, "character_name")}
-        </Box>
-      </Box>
-      {map(keys(omit(charBuild, "character_name")), (stats, idx) => {
-        return (
-          <Box key={idx} style={{ display: "flex", alignItems: "center" }}>
-            <Box
-              style={{
-                justifyContent: "flex-start",
-                minWidth: "250px",
-                textAlign: "right"
-              }}
-              marginRight="20px"
-            >
-              {stats}
-            </Box>
-            <Box style={{ justifyContent: "flex-end", minWidth: "100px" }}>
-              <TextField
-                type="number"
-                label={stats}
-                inputProps={{
-                  style: {
-                    color: "white",
-                    fontFamily: "genshinFont"
-                  }
-                }}
-                size="small"
-                className={classes.field}
-                onChange={e => (charBuild[stats] = parseFloat(e.target.value))}
-                defaultValue={get(charBuild, stats)}
-              ></TextField>
-            </Box>
-          </Box>
-        );
-      })}
-      {map(keys(baseDamage), (talent, idx) => {
+    <Box>
+      {map(STAT_FIELDS, (stats, idx) => {
         return (
-          <Box key={idx} style={{ display: "flex" }}>
-            <Box style={{ justifyContent: "flex-start", minWidth: "250px" }}>
-              {talent}
-            </Box>
-            <Box style={{ justifyContent: "flex-end", minWidth: "100px" }}>
-              {get(get(baseDamage, talent), "damage")}
-            </Box>
+          <Box key={`${stats}${idx}`}>
+            {!(stats.modifiable ?? true) && (
+              <Box style={{ display: "flex", alignItems: "center" }}>
+                <Box
+                  style={{
+                    justifyContent: "flex-start",
+                    minWidth: "250px",
+                    textAlign: "right",
+                  }}
+                  marginRight="20px"
+                >
+                  {stats.label}
+                </Box>
+                <Box style={{ justifyContent: "flex-end", minWidth: "100px" }}>
+                  {get(props.characterBuild, "character_name")}
+                </Box>
+              </Box>
+            )}
+            {(stats.modifiable ?? true) && (
+              <Box key={idx} style={{ display: "flex", alignItems: "center" }}>
```

Comment:
maybe you needed a key here but i wasnt sure why this line, line 44 had a key index but line 27 did not have one, which is the box without modify

Reply:
didnt need a key here actually, bc its a nested component inside the actual mapped box
good catch